tchannel/tchannel.py

```diff
@@ -232,6 +236,12 @@ def advertise(self, routers, name=None, timeout=None):
             The timeout (in seconds) for the initial advertise attempt.
             Defaults to 30 seconds.
 
+        :param router_file:
+            The host file that contains the routers information. The file
+            should contain a JSON stringified format of the routers parameter.
+            Either routers or router_file should be provided. If both provided,
+            routers will be used.
```

Comment:
We haven't yet defined an exception specific to invalid parameters. Just check if they're both None or if they're both not None and raise a `ValueError` with a message like "Only one of routers and router_file must be provided."

I'm fairly sure we just have assertions in some part of the code, but generally a bad idea for public APIs.